README.md

```diff
@@ -22,16 +22,37 @@ Flatpak welcomes contributions from anyone! Here are some ways you can help:
 * Improve the [Flatpak support](https://github.com/flatpak/flatpak/wiki/Distribution) in your favorite Linux distribution
 
 # Hacking
-Flatpak uses a traditional autoconf-style build mechanism. To build just do
+Flatpak uses a traditional autoconf-style build mechanism. The exact steps
+required depend on your distribution. Here are some steps that should work on
+Debian and Fedora:
+
+## On Debian
+```
+git clone https://github.com/flatpak/flatpak
+cd flatpak
+sudo apt build-dep flatpak
+NOCONFIGURE=1 ./autogen.sh
+./configure --build=x86_64-linux-gnu --prefix=/usr --includedir=\${prefix}/include --mandir=\${prefix}/share/man --infodir=\${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --disable-option-checking --disable-silent-rules --libdir=\${prefix}/lib/x86_64-linux-gnu --runstatedir=/run --disable-maintainer-mode --disable-dependency-tracking --disable-auto-sideloading --disable-selinux-module --enable-gdm-env-file --enable-installed-tests --with-dbus-config-dir=/usr/share/dbus-1/system.d --with-priv-mode=none --with-privileged-group=sudo --with-run-media-dir=/media --with-system-bubblewrap=bwrap --with-system-dbus-proxy=xdg-dbus-proxy --with-systemdsystemunitdir=/lib/systemd/system --with-system-helper-user=_flatpak --enable-docbook-docs --enable-documentation --disable-gtk-doc
+make -j$(nproc)
+make check -j$(nproc)
+sudo make install
+```
+
+## On Fedora
+
 ```
- ./autogen.sh
- ./configure [args]
- make
- make install
+git clone https://github.com/flatpak/flatpak
+cd flatpak
+sudo dnf builddep flatpak
+sudo dnf install gettext-devel socat
+NOCONFIGURE=1 ./autogen.sh
+./configure --build=x86_64-redhat-linux-gnu --host=x86_64-redhat-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --enable-docbook-docs --enable-installed-tests --enable-selinux-module --with-priv-mode=none --with-system-bubblewrap --with-system-dbus-proxy
```

Comment:
Similar considerations here. I don't think Flatpak uses `--sharedstatedir=/var/lib`, so that one is probably redundant?

Reply:
Yeah I think you're right